packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,143 @@
+      loadTestFileToJsObject(('child-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStackTemplate('ChildStack').getNestedStack('GrandChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('grandchild-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStack('AnotherChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('another-child-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStackTemplate('AnotherChildStack').getNestedStack('GrandChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('grandchild-stack.expected.json')),
+    );
+  });
+
+  test("throws an error when NestedStacks contain a stack not found in the template", () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: _testTemplateFilePath('parent-export-stack.json'),
+        nestedStacks: {
+          "FakeStack": {
+            templateFile: _testTemplateFilePath('child-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/Nested Stack with logical ID 'FakeStack' was not found in the template/);
+  });
+
+  test("Throws an error when NestedStacks contains an ID that isn't a CloudFormation::Stack in the template", () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: _testTemplateFilePath('parent-export-stack.json'),
+        nestedStacks: {
+          "ChildStack": {
+            templateFile: _testTemplateFilePath('child-import-stack.json'),
+            nestedStacks: {
+              "BucketImport": {
+                templateFile: _testTemplateFilePath('grandchild-import-stack.json'),
+              },
+            },
+          },
+        },
+      });
+    }).toThrow(/Nested Stack with logical ID 'BucketImport' is not an AWS::CloudFormation::Stack resource/);
+  });
+
+  test("can modify resources in nested stacks", () => {
+    const parent = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: _testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        "ChildStack": {
+          templateFile: _testTemplateFilePath('child-import-stack.json'),
+          nestedStacks: {
+            "GrandChildStack": {
+              templateFile: _testTemplateFilePath('grandchild-import-stack.json'),
+            },
+          },
+        },
+      },
+    });
+
+    const grandChildTemplate = parent.getNestedStackTemplate('ChildStack').getNestedStackTemplate('GrandChildStack');
+    const bucket = grandChildTemplate.getResource('BucketImport') as s3.CfnBucket;
+
+    bucket.bucketName = 'modified-bucket-name';
+
+    const grandChildStack = parent.getNestedStackTemplate('ChildStack').getNestedStack('GrandChildStack');
+    expect(grandChildStack).toHaveResource('AWS::S3::Bucket', { BucketName: 'modified-bucket-name' });
+  });
+
+  test("templates with nested stacks that don't exist in the nested stacks property have those stacks ignored", () => {
```

Comment:
This test should simply not pass the `nestedStacks` property to `CfnInclude`, and assert the result template is the same as the input template.